mate-panel/panel-toplevel.c

```diff
@@ -2931,15 +2931,23 @@ panel_toplevel_move_resize_window (PanelToplevel *toplevel,
 			AppletInfo *info;
 			info = mate_panel_applet_get_by_id (id);
 
-			stick = g_settings_get_boolean (info->settings, PANEL_OBJECT_PANEL_RIGHT_STICK_KEY);
+			edge_relativity = g_settings_get_enum (info->settings, PANEL_OBJECT_RELATIVE_TO_EDGE_KEY);
 
-			if (stick) {
+			if (edge_relativity == PANEL_EDGE_CENTER ||
+			    edge_relativity == PANEL_EDGE_END) {
 				int position = g_settings_get_int (info->settings, PANEL_OBJECT_POSITION_KEY);
-				if (toplevel->priv->orientation & PANEL_HORIZONTAL_MASK) {
-					ad->pos = toplevel->priv->geometry.width - position;
-				} else {
-					ad->pos = toplevel->priv->geometry.height - position;
-				}
+				int size = mate_panel_applet_get_size (info);
+				int toplevel_size;
+
+				if (toplevel->priv->orientation & PANEL_HORIZONTAL_MASK)
+					toplevel_size = toplevel->priv->geometry.width;
+				else
+					toplevel_size = toplevel->priv->geometry.height;
+
+				if (edge_relativity == PANEL_EDGE_CENTER)
+					ad->pos = toplevel_size/2 - size/2 + position;
```

Comment:
cast to `int`

Reply:
Ummm...  Aren't we only working with integers here?

 - `toplevel_size`:  Declared `int` at `mate-panel/panel-toplevel.c` @ 2940.
 - `size`:  Declared `int` at `mate-panel/panel-toplevel.c` @ 2939.
 - `position`:  Declared `int` at `mate-panel/panel-toplevel.c` @ 2938.

I see literally no reason to include an explicit cast.  Am I missing something here?